Inventory.cs

```diff
@@ -3,34 +3,42 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Xml.Linq;
 
 namespace SimpleInv
 {
     public class Inventory : IInventory
     {
+        private List<Product> _products;
+
+        public Inventory() {
+            _products = new List<Product>();
+        }
         public void AddProduct(Product product)
         {
-            throw new NotImplementedException();
+            _products.Add(product);
         }
 
         public void RemoveProduct(Product product)
         {
-            throw new NotImplementedException();
+            _products.Remove(product);
         }
 
         public Product Retrieve(string productName)
         {
-            throw new NotImplementedException();
+            return _products.Find(x => x.Name.Equals(productName, StringComparison.OrdinalIgnoreCase));
         }
 
         public List<Product> Retrieve()
         {
-            throw new NotImplementedException();
+            return _products;
         }
 
         public void UpdateProduct(Product product, string newName, decimal newPrice, decimal newQuantity)
```

Comment:
Just worndering about the update method here.
Usually I'd expect to pass and object that represent the updated Product to the Inventory update method which will reflect that to the In-Memery/DB product. and not passing each property we wanna update to the update method.